manim/mobject/svg/text_mobject.py

```diff
@@ -1385,6 +1401,9 @@ def extract_color_tags(self):
         self.text = re.sub("<color[^>]+>(.+?)</color>", r"\1", self.text, 0, re.S)
         return colormap
 
+    def init_colors(self, propagate_colors=True):
+        SVGMobject.init_colors(self, propagate_colors=propagate_colors)
```

Comment:
TODO: Remove the colour methods present in `MarkupText` and directly translate it to Pango.
/cc @PhilippImhof

Reply:
I haven't had the time to dig into the details of this PR. But if we get the SVG directly from Pango, together **with** the color (which we did not until now), then we could just use `<span foreground="#003355" >` and the like.

However, we still need to deal with gradients (unless we want to abandon that feature, but why should we) and we would lose the ability to use the predefined color constants.

What do we gain from having the color done in Pango as opposed to how we do it now?

Reply:
> What do we gain from having the color done in Pango as opposed to how we do it now?

Things will be aligned and mapped correctly.

> However, we still need to deal with gradients (unless we want to abandon that feature, but why should we) and we would lose the ability to use the predefined color constants.

> However, we still need to deal with gradients (unless we want to abandon that feature, but why should we) and we would lose the ability to use the predefined color constants.

yeah, gradients... I am gonna make an issue in Pango Gitlab asking for this feature.

This should anyhow need to be documented but that's only after this PR's merge.

Reply:
Gradients are also not yet supported by manim, right?

It would also be more 'elegant' to have the colors as part of the SVG file, now that it's possible for Manim to read those colors. Then the file that Pango produces could be a stronger 'source of truth' - if that makes sense. I don't like how right now only some of the SVG information is carried over, and some parts are set in a different way. But that's minor in the grand scheme of things.

Reply:
> Gradients are also not yet supported by manim, right?

It is supported. see https://docs.manim.community/en/v0.3.0/reference/manim.mobject.svg.text_mobject.MarkupText.html?highlight=markuptext#colorexample

Reply:
Is that implemented so that each letter is its own color, and that color is along a gradient? Or, for example, can a single letter have a gradient?

Reply:
> Is that implemented so that each letter is its own color, and that color is along a gradient? Or, for example, can a single letter have a gradient?

Currently, single letters cannot have a gradient.
```python
from manim import *
from manimpango import *


class ColorExample(Scene):
    def construct(self):
        text1 = MarkupText(
            '<color col="YELLOW">a</color><color col="RED">b</color>'
        )
        text2 = MarkupText("nice gradient", gradient=(BLUE, YELLOW))
        text3 = MarkupText(
            '<gradient from="#ff0000" to="#00ff00">x</gradient><gradient from="BLUE" to="GREEN">y</gradient>', size=2
        )
        text4 = MarkupText("a", gradient=(BLUE, YELLOW), size=2)
        group = VGroup(text1, text2, text3, text4).arrange(DOWN)
        self.add(group)
```
![ColorExample](https://user-images.githubusercontent.com/52650214/107887621-9fb7a900-6f07-11eb-905c-c2c668307442.png)

@naveen521kk 
> Things will be aligned and mapped correctly.

You mean the occasional problems with ligatures and offsets?

Reply:
> It would also be more 'elegant' to have the colors as part of the SVG file, now that it's possible for Manim to read those colors. 

I fully agree. I did not do manual coloring out of choice, but simply because we could not have them via the SVG. I think `<span foreground="...">bla</span>` will do the trick, as this is officially supported by Pango's markup language. I will try to test that next week.

Reply:
> You mean the occasional problems with ligatures and offsets?

yeah

Reply:
also https://gitlab.gnome.org/GNOME/pango/-/issues/533